pkg/api/ldap_debug_test.go

```diff
@@ -379,8 +358,151 @@ func TestGetLDAPStatusApiEndpoint(t *testing.T) {
 		{ "host": "10.0.0.5", "port": 361, "available": false, "error": "something is awfully wrong" }
 	]
 	`
-	var expectedJSON interface{}
-	_ = json.Unmarshal([]byte(expected), &expectedJSON)
+	assert.JSONEq(t, expected, sc.resp.Body.String())
+}
+
+//***
+// PostSyncUserWithLDAP tests
+//***
+
+func postSyncUserWithLDAPContext(t *testing.T, requestURL string) *scenarioContext {
+	t.Helper()
+
+	sc := setupScenarioContext(requestURL)
+
+	ldap := setting.LDAPEnabled
+	setting.LDAPEnabled = true
+	defer func() { setting.LDAPEnabled = ldap }()
+
+	hs := &HTTPServer{Cfg: setting.NewCfg()}
+
+	sc.defaultHandler = Wrap(func(c *models.ReqContext) Response {
+		sc.context = c
+		return hs.PostSyncUserWithLDAP(c)
+	})
+
+	sc.m.Post("/api/admin/ldap/sync/:id", sc.defaultHandler)
+
+	sc.resp = httptest.NewRecorder()
+	req, _ := http.NewRequest(http.MethodPost, requestURL, nil)
+	sc.req = req
+	sc.exec()
+
+	return sc
+}
+
+func TestPostSyncUserWithLDAPAPIEndpoint_Success(t *testing.T) {
+	getLDAPConfig = func() (*ldap.Config, error) {
+		return &ldap.Config{}, nil
+	}
+
+	newLDAP = func(_ []*ldap.ServerConfig) multildap.IMultiLDAP {
+		return &LDAPMock{}
+	}
+
+	userSearchResult = &models.ExternalUserInfo{
+		Login: "ldap-daniel",
+	}
+
+	bus.AddHandler("test", func(cmd *models.UpsertUserCommand) error {
```

Comment:
I thought you [didn't like](https://github.com/grafana/grafana/pull/18365#discussion_r310236114) the `test` names? :)